lib/flake.nix

```
@@ -0,0 +1,72 @@
+{
+  description = "DevOS environment configuriguration library.";
+
+  inputs =
+    {
+      deploy.url = "github:serokell/deploy-rs";
+      deploy.inputs = {
+        utils.follows = "flake-utils";
+      };
+      devshell.url = "github:numtide/devshell";
+      flake-utils.url = "github:numtide/flake-utils";
+      
+    };
+
+  outputs = inputs@{ self, nixpkgs, deploy, devshell, flake-utils, ... }: let
+
+    inherit (nixpkgs) lib;
+
+    # We work with a combined lib, internally ...
+    combinedLib = lib.extend (final: prev:
+      let callLibs = file: import file
+        ({
+          lib = final;
```

```suggestion
          lib = lib // { dev = final; }
```
And use lib.makeExtensible. I think thats clearer and helps towards removing callLibs. Also so we don't have to manually export things. And I think we should export all of devos lib, and only manually create the top level functions. Just don't want nixpkgs lib to be exported.

---

Although I guess this still keeps the lib vs dev issue. Then maybe we should just do `lib = recursiveUpdate lib final`